tensorflow_probability/python/distributions/empirical_test.py

```diff
@@ -0,0 +1,694 @@
+        [0],
+        [0, 1],
+        [[0, 1], [1, 2]]
+    ]
+    expected_pmfs = [
+        [0.25],
+        [0.25, 0.5],
+        [[0.25, 0.5], [0.5, 0.25]]
+    ]
+
+    for event, expected_pmf in zip(events, expected_pmfs):
+      input_ = tf.convert_to_tensor(sample, dtype=np.float32)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=input_.shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph)
+      self.assertAllClose(self.evaluate(dist.prob(event)),
+                          expected_pmf)
+
+  def testPmfWithBatch(self):
+    sample = [[0, 0, 1, 2], [0, 1, 1, 2]]
+    events = [
+        [0],
+        [0, 1],
+        [[0, 1], [1, 2]]
+    ]
+    expected_pmfs = [
+        [0.5, 0.25],
+        [0.5, 0.5],
+        [[0.5, 0.5], [0.25, 0.25]]
+    ]
+
+    for event, expected_pmf in zip(events, expected_pmfs):
+      input_ = tf.convert_to_tensor(sample, dtype=np.float32)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=input_.shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph)
+      self.assertAllClose(self.evaluate(dist.prob(event)),
+                          expected_pmf)
+
+  def testPmfInvalid(self):
+    dist = empirical.Empirical(samples=random_samples([2, 2]),
+                               validate_args=True)
+    invalid_events = [
+        [],
+        [0, 0, 1]
+    ]
+    for event in invalid_events:
+      with self.assertRaises(ValueError):
+        self.evaluate(dist.prob(event))
+
+  def testEntropy(self):
+    samples = [
+        [0],
+        [0, 1, 0, 2, 2],
+        [[0, 0], [0, 1], [1, 2]],
+        [[[0, 0, 0, 1], [0, 0, 1, 1]],
+         [[1, 1, 1, 2], [0, 1, 2, 2]]]
+    ]
+    expected_entropys = [
+        entropy([1]),
+        entropy([0.4, 0.2, 0.4]),
+        [entropy([1]), entropy([0.5, 0.5]), entropy([0.5, 0.5])],
+        [[entropy([0.75, 0.25]), entropy([0.5, 0.5])],
+         [entropy([0.75, 0.25]), entropy([0.25, 0.25, 0.5])]]
+    ]
+
+    for sample, expected_entropy in zip(samples, expected_entropys):
+      input_ = tf.convert_to_tensor(sample, dtype=np.float64)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=input_.shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph)
+      self.assertAllClose(self.evaluate(dist.entropy()), expected_entropy)
+
+  def testSampleN(self):
+    for samples_shape in ([2], [2, 4], [4, 2, 4]):
+      input_ = random_samples(samples_shape)
+      input_ph = tf.placeholder_with_default(
+          input=input_,
+          shape=samples_shape if self.static_shape else None)
+      dist = empirical.Empirical(samples=input_ph)
+      self.assertAllEqual(self.evaluate(tf.shape(dist.sample())),
+                          dist.batch_shape_tensor())
+
+      n = 1000
+      seed = 42
+      self.assertAllEqual(self.evaluate(dist.sample(n, seed)),
```

Comment:
this won't work well in eager mode (can you add `@tfe.run_test_in_graph_and_eager_mode` to the test class? see https://github.com/tensorflow/probability/blob/master/UNITTEST.md#tfp-unit-test-best-practices)

With eager, you might be able to do
```python
seed=tf.set_random_seed(42) if tf.executing_eagerly() else 42
samples1 = dist.sample(n, seed)
seed=tf.set_random_seed(42) if tf.executing_eagerly() else 42
samples2 = dist.sample(n, seed)
self.assertAllEqual(self.evaluate(samples1), self.evaluate(samples2))
```

Reply:
You are right. It seems that previously when I invoked (tensorflow.python.framework) `tf_test_util.run_all_in_graph_and_eager_modes` before test class, it failed to run in eager mode...